sources/Launcher/DYNRobustnessAnalysisLauncher.cpp

```diff
@@ -129,6 +143,82 @@ RobustnessAnalysisLauncher::init() {
   multipleJobs_ = readInputData(fileName);
 }
 
+struct Version {
+  Version() {}
+  Version(std::string n, std::string v, std::string b, std::string h) :
+    project_name(n),
+    version_string(v),
+    git_branch(b),
+    git_hash(h) {
+  }
+  std::string project_name;
+  std::string version_string;
+  std::string git_branch;
+  std::string git_hash;
+};
+
+void
+RobustnessAnalysisLauncher::initLog() {
+  std::vector<Trace::TraceAppender> appenders;
+  Trace::TraceAppender appender;
+  std::string outputPath(createAbsolutePath("dynawo.log", workingDirectory_));
+
+  appender.setFilePath(outputPath);
+#if _DEBUG_
+  appender.setLvlFilter(DYN::DEBUG);
+#else
+  appender.setLvlFilter(DYN::INFO);
+#endif
+  appender.setTag(logTag_);
+  appender.setShowLevelTag(true);
+  appender.setSeparator(" | ");
+  appender.setShowTimeStamp(true);
+  appender.setTimeStampFormat("%Y-%m-%d %H:%M:%S");
+  appender.setPersistant(true);
+
+  appenders.push_back(appender);
+
+  Trace::addAppenders(appenders);
+
+  // known projects versions
+  std::vector<Version> versions;
+  versions.push_back(Version("Dynawo", DYNAWO_VERSION_STRING, DYNAWO_GIT_BRANCH, DYNAWO_GIT_HASH));
+  versions.push_back(Version("Dynawo-algorithms", DYNAWO_ALGORITHMS_VERSION_STRING, DYNAWO_ALGORITHMS_GIT_BRANCH, DYNAWO_ALGORITHMS_GIT_HASH));
+  std::set<std::string> projects;
+  projects.insert("DYNAWO");
+  projects.insert("DYNAWO-ALGORITHMS");
+
+  // get additional projects versions in versions.ini if exists
+  std::string versions_ini = (boost::dll::program_location().parent_path() / "versions.ini").string();
+  if (exists(versions_ini)) {
+    namespace pt = boost::property_tree;
+    pt::iptree tree;
+    try {
+      pt::read_ini(versions_ini, tree);
+      BOOST_FOREACH(pt::iptree::value_type &section, tree) {
```

Comment:
I would expect an update on private side to use this

Reply:
It's in progress...